pkg/network/usm/monitor_test.go

```diff
@@ -915,6 +917,171 @@ func (s *USMHTTP2Suite) TestHTTP2KernelTelemetry() {
 	}
 }
 
+func (s *USMHTTP2Suite) TestRawTraffic() {
+	t := s.T()
+	cfg := networkconfig.New()
+	cfg.EnableHTTP2Monitoring = true
+
+	startH2CServer(t)
+
+	tests := []struct {
+		name              string
+		runClients        func(t *testing.T, clientsCount int)
+		expectedEndpoints map[http.Key]captureRange
+	}{
+		{
+			name: "test parse frames tail call using raw tcp",
+			// The purpose of this test is to validate that when we surpass the limit of HTTP2_MAX_FRAMES_ITERATIONS,
+			// the filtering of subsequent frames will continue using tail calls.
+			// Raw TCP is necessary for sending a large number of frames.
+			runClients: func(t *testing.T, clientsCount int) {
+				magicFrame := []byte{
+					// http2 magic
+					0x50, 0x52, 0x49, 0x20, 0x2a, 0x20, 0x48, 0x54, 0x54, 0x50, 0x2f, 0x32, 0x2e, 0x30, 0x0d, 0x0a, 0x0d, 0x0a, 0x53, 0x4d, 0x0d, 0x0a, 0x0d, 0x0a,
+				}
+				settingsFrame := []byte{
+					// settings frame
+					0x00, 0x00, 0x12, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04, 0x00, 0x40, 0x00, 0x00, 0x00, 0x06, 0x00, 0xa0, 0x00, 0x00,
+				}
```

Comment:
we can use a smaller (probably empty) settings frame if we want to reduce payload's size